pkg/cache/application.go

```diff
@@ -81,8 +87,14 @@ func NewApplication(appID, queueName, user string, tags map[string]string, sched
 			{Name: string(events.AcceptApplication),
 				Src: []string{states.Submitted, states.Recovering},
 				Dst: states.Accepted},
+			{Name: string(events.TryReserve),
+				Src: []string{states.Accepted},
+				Dst: states.Reserving},
+			{Name: string(events.UpdateReservation),
+				Src: []string{states.Reserving},
+				Dst: states.Reserving},
```

Comment:
hi @kingamarton  pls take a look at the SM definition here:

```
fsm.Callbacks{
  ...
  string(events.UpdateReservation):      app.onReservationStateChange,
  ...
},
```

this is called every time when a task is bound to a node, we check if the app gets enough placeholders; when it gets enough, in `onReservationStateChange()` it will trigger a state transition from `Reserving` to `Running`.

```
func (app *Application) onReservationStateChange(event *fsm.Event) {
         ...
	if desireCounts.Equals(actualCounts) {
		ev := NewRunApplicationEvent(app.applicationID)
		dispatcher.Dispatch(ev)
	}
}
```